### PR TITLE
Fix couple issues in subprojects update command

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

File: mesonbuild/msubprojects.py [hunk 1]
```diff
@@ -441,6 +441,7 @@ def update(self) -> bool:
             success = self.update_svn()
         elif self.wrap.type is None:
             self.log('  -> Cannot update subproject with no wrap file')
+            success = True
```
Comment (eli-schwartz) [inline mesonbuild/msubprojects.py:444]: I think we should only do this if we are updating "all the wraps", but still consider it an error if a single wrap was passed on the command line and it cannot be updated.

File: mesonbuild/msubprojects.py [hunk 2]
```diff
@@ -444,7 +444,7 @@ def update(self) -> bool:
             success = True
         else:
             self.log('  -> Cannot update', self.wrap.type, 'subproject')
-        if success:
+        if success and os.path.exists(self.wrap_resolver.dirname):
```
Comment (eli-schwartz) [inline mesonbuild/msubprojects.py:447]: This is only an issue if we consider "cannot update subproject with no wrap file" to be success.

Hence it is part of the same commit IMO. But, I think we should check that more directly by skipping this in the case that we observed there is no wrap file. Maybe that should be the condition, even. Or maybe we should track this with `skipped = False`, and finally `return success or skipped`.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/msubprojects.py (hunk 2 above)
Comment (xclaesse) [reply mesonbuild/msubprojects.py:447]: No, it happens when updating all wraps and some have not been downloaded.

Comment (xclaesse) [reply mesonbuild/msubprojects.py:447]: update_file() returns success if dir does not exists.

File: mesonbuild/msubprojects.py (hunk 1 above)
Comment (xclaesse) [reply mesonbuild/msubprojects.py]: fair enough, changed

File: mesonbuild/msubprojects.py (hunk 2 above)
Comment (xclaesse) [reply mesonbuild/msubprojects.py]: Changed to consider it success only when updating all subprojects.

**Pull request merged**